nexus_constructor/component.py

```diff
@@ -1,13 +1,83 @@
 import attr
-from nexus_constructor.transformations import Transformation
+import h5py
+from typing import Any, List
 from nexus_constructor.pixel_data import PixelData
 from nexus_constructor.geometry_types import Geometry
-from typing import List
+from nexus_constructor.nexus import nexus_wrapper as nx
+from nexus_constructor.transformations import Transformation, create_transform
+
+
+class ComponentModel:
+    """
+    Provides an interface to an existing component group in a NeXus file
+    """
+
+    def __init__(self, nexus_file: nx.NexusWrapper, group: h5py.Group):
+        self.file = nexus_file
+        self.group = group
+
+    @property
+    def name(self):
+        return nx.get_name_of_group(self.group)
+
+    @name.setter
+    def name(self, new_name: str):
+        self.file.rename_group(self.group, new_name)
+
+    def get_field(self, name: str):
+        return self.file.get_field_value(self.group, name)
+
+    def set_field(self, name: str, value: Any, dtype=None):
+        self.file.set_field_value(self.group, name, value, dtype)
+
+    @property
+    def nx_class(self):
+        return self.file.get_nx_class(self.group)
+
+    @nx_class.setter
+    def nx_class(self, nx_class: str):
+        self.file.set_nx_class(self.group, nx_class)
+
+    @property
+    def description(self):
+        return self.file.get_field_value(self.group, "description")
+
+    @description.setter
+    def description(self, description: str):
+        if description:
+            self.file.set_field_value(self.group, "description", description, str)
+
+    @property
+    def transforms(self):
+        """
+        Gets all transforms in the depends_on chain for this component
+        :return: List of transforms
+        """
+        transforms = []
```

Comment:
It would be nice if the list is updated with new transformations instead of being re-created every time you call `transforms()`. With that said, it is probably not that hard to work around this.

Reply:
To reduce complexity I'm trying to avoid that there is any cached data layer between the in-memory NeXus file and the GUI.